data/campaigns/Eastern_Invasion/scenarios/04b_Ill_Humours.cfg

```diff
@@ -0,0 +1,1055 @@
+        image=items/swamp_herb.png
+        visible_in_fog=no
+    [/item]
+    [event]
+        name=moveto
+        [filter]
+            side=1
+            x,y={X},{Y}
+        [/filter]
+        [modify_unit]
+            [filter]
+                id=$unit.id
+            [/filter]
+            {HERB_OVERLAY_ONESCENARIO}
+            [status]
+                unpoisonable=yes
+            [/status]
+        [/modify_unit]
+
+        [if]
+            [have_unit]
+                id=Hahid al-Ali
+            [/have_unit]
+
+            [and]
+                {VARIABLE_CONDITIONAL herbs_needed greater_than 0}
+            [/and]
+
+            [then]
+                {VARIABLE_OP herbs_needed sub 1}
+                [if]
+                    {VARIABLE_CONDITIONAL herbs_needed less_than_equal_to 0}
+                    [then]
+                        {REMOVE_IMAGE {X} {Y}}
+                        [fire_event]
+                            name=herbs_collected
+                        [/fire_event]
+                    [/then]
+
+                    [else]
+                        [message]
+                            speaker=unit
+                            message= _ "I’ve found a sprig of alswdan and am now immune to the swamp’s poison! We’ll need $herbs_needed more (and $gold_cost gold) to travel east."
+                        [/message]
+                    [/else]
+                [/if]
+            [/then]
+
+            [else]
+                [message]
+                    speaker=unit
+                    message= _ "I’ve found a sprig of alswdan and am now immune to the swamp’s poison!"
+                [/message]
+            [/else]
+        [/if]
+
+        {REMOVE_IMAGE {X} {Y}}
+    [/event]
+#enddef
+#define RANDOM_HERB I X Y
+    [random_placement]
+        variable=location
+        num_items=1
+        [filter_location]
+            x,y={X},{Y}
+            terrain=Ss,Sm,Sm^Qhu
+        [/filter_location]
+        [command]
+            {PLACE_HERB $location.x $location.y}
+            {VARIABLE herb_locations[{I}].x $location.x}
+            {VARIABLE herb_locations[{I}].y $location.y}
+        [/command]
+    [/random_placement]
+#enddef
+
+    [event]
+        name=place_herbs
+        {RANDOM_HERB 1  37-45  1-4 } # top right corner (hard)
+        {RANDOM_HERB 2  34-42  5-10} # upper- middle right (medium)
+        {RANDOM_HERB 3  34-42 11-16} # middle-middle right (medium)
+        {RANDOM_HERB 4  34-42 17-22} # lower- middle right (medium)
+        {RANDOM_HERB 5  37-45 23-25} # bottom right corner (hard)
+
+        {RANDOM_HERB 6   6-11 11-15} # near the mountains (hard)
+        {RANDOM_HERB 7   8-18 15-24} # near the dunefolk (easy)
+
+        {RANDOM_HERB 8  28-36 17-22} # southeast of Gweddry (easy)
+        {RANDOM_HERB 9  33-35  8-11} # northeast of Gweddry (easy)
+        {RANDOM_HERB 10 18-29  1-3 } # north of Gweddry (medium)
+        {RANDOM_HERB 11 18-29  5-8 } # north of Gweddry (medium)
+    [/event]
+
```

Comment:
The events seem to not be firing. 
[EI-Ill Humours replay 20230831-230037.gz](https://github.com/wesnoth/wesnoth/files/12490559/EI-Ill.Humours.replay.20230831-230037.gz)

Here is a suggestion that worked for me:

```suggestion
#define PLACE_HERB I
    [item]
        x,y=$herb_locations[{I}].x, $herb_locations[{I}].y
        image=items/swamp_herb.png
        visible_in_fog=no
    [/item]
    [message]
        speaker = narrator
        message = "Placing herb in $herb_locations[{I}].x, $herb_locations[{I}].y"
    [/message]
    [event]
        name=moveto
        [filter]
            side=1
            x,y=$herb_locations[{I}].x, $herb_locations[{I}].y
        [/filter]
        [modify_unit]
            [filter]
                id=$unit.id
            [/filter]
            {HERB_OVERLAY_ONESCENARIO}
            [status]
                unpoisonable=yes
            [/status]
        [/modify_unit]

        [if]
            [have_unit]
                id=Hahid al-Ali
            [/have_unit]

            [and]
                {VARIABLE_CONDITIONAL herbs_needed greater_than 0}
            [/and]

            [then]
                {VARIABLE_OP herbs_needed sub 1}
                [if]
                    {VARIABLE_CONDITIONAL herbs_needed less_than_equal_to 0}
                    [then]
                        {REMOVE_IMAGE $herb_locations[{I}].x, $herb_locations[{I}].y}
                        [fire_event]
                            name=herbs_collected
                        [/fire_event]
                    [/then]

                    [else]
                        [message]
                            speaker=unit
                            message= _ "I’ve found a sprig of alswdan and am now immune to the swamp’s poison! We’ll need $herbs_needed more (and $gold_cost gold) to travel east."
                        [/message]
                    [/else]
                [/if]
            [/then]

            [else]
                [message]
                    speaker=unit
                    message= _ "I’ve found a sprig of alswdan and am now immune to the swamp’s poison!"
                [/message]
            [/else]
        [/if]

        {REMOVE_IMAGE $herb_locations[{I}].x, $herb_locations[{I}].y}
    [/event]
#enddef
#define RANDOM_HERB I X Y
    [random_placement]
        variable=location
        num_items=1
        [filter_location]
            x,y={X},{Y}
            terrain=Ss,Sm,Sm^Qhu
        [/filter_location]
        [command]
            {VARIABLE herb_locations[{I}].x $location.x}
            {VARIABLE herb_locations[{I}].y $location.y}
            {PLACE_HERB {I}}
        [/command]
    [/random_placement]
#enddef

    [event]
        name=place_herbs
        {RANDOM_HERB 1  37-45  1-4 } # top right corner (hard)
        {RANDOM_HERB 2  34-42  5-10} # upper- middle right (medium)
        {RANDOM_HERB 3  34-42 11-16} # middle-middle right (medium)
        {RANDOM_HERB 4  34-42 17-22} # lower- middle right (medium)
        {RANDOM_HERB 5  37-45 23-25} # bottom right corner (hard)

        {RANDOM_HERB 6   6-11 11-15} # near the mountains (hard)
        {RANDOM_HERB 7   8-18 15-24} # near the dunefolk (easy)

        {RANDOM_HERB 8  28-36 17-22} # southeast of Gweddry (easy)
        {RANDOM_HERB 9  33-35  8-11} # northeast of Gweddry (easy)
        {RANDOM_HERB 10 18-29  1-3 } # north of Gweddry (medium)
        {RANDOM_HERB 11 18-29  5-8 } # north of Gweddry (medium)
    [/event]

```

Reply:
Thanks for the fix!